### PR TITLE
Remove unneeded iamRoleArn property

### PR DESCRIPTION
**Pull request by svdgraaf**
As mentioned in issue https://github.com/serverless/serverless/issues/1469 We don't need the iamRoleArn property. We can have CloudFormation generate it dynamically for us, that's what this PR will do :smiley:

### CODE REVIEW

**Comment by eahefnawy** [comment]
Thank you kind sir! 😊 ... that is awesome! 💃

**Comment by pmuens** [comment]
Wow @svdgraaf you rock! Thanks for that! 🎉

**Comment by svdgraaf** [comment]
I guess this also fixes #1566?

**Comment by eahefnawy** [comment]
@svdgraaf one tiny fix. Now that we no longer use this variable, could you also update the deploy plugin to not add the variable to `serverless.env.yaml` after deployment? 😊 ... otherwise we'd be saving sensitive data that are not necessary

**Comment by svdgraaf** [comment]
@eahefnawy sure, on it :)

**Comment by svdgraaf** [comment]
@eahefnawy got it. I removed the reference from the initial stack output.

**Comment by eahefnawy** [comment]
@svdgraaf that's great! ... could you also remove the [addOutputVars method](https://github.com/serverless/serverless/blob/v1.0/lib/plugins/aws/deploy/lib/createStack.js#L76-L98) and update the relavent tests 😊 ?

**Comment by svdgraaf** [comment]
Yes. I didn't remove it because I thought the bucket var might still be in use. I'll look into it.

Groeten!
Sander

**Comment by pmuens** [comment]
@svdgraaf we extract the deploymentBucketName here (so you can remove it from my side): https://github.com/serverless/serverless/blob/v1.0/lib/plugins/aws/index.js#L91-L104

Other than that I would add the `console.log` which prints out that the stack was successfully created to another function so that the user still gets the info that the stack was successfully created.

**Comment by svdgraaf** [comment]
@pmuens ok, will do!

**Comment by pmuens** [comment]
@svdgraaf Thanks for that! 🎉 

There are just some minor linting issues. Could you maybe rebase the branch additionally? We've updated our integration test yesterday which was failing all the time.

Thanks in advance

**Comment by svdgraaf** [comment]
@pmuens yeah, I was still working on it :) will give a headsup when it's done :)

**Comment by svdgraaf** [comment]
I think I'll create a new PR after I have everything in place, so that we can get a cleaner history :)

File: lib/plugins/aws/deploy/compile/events/apiGateway/tests/methods.js [hunk 1]
```diff
@@ -163,25 +163,23 @@ describe('#compileMethods()', () => {
...
-        ]
-      ]
-    }
+      'Fn::Join': ['', [
+        'arn:aws:apigateway:',
+        { Ref: 'AWS::Region' },
+        ':lambda:path/2015-03-31/functions/',
+        { 'Fn::GetAtt': ['first', 'Arn'] },
+        '/invocations'  // eslint-disable-line comma-dangle
+      ]]                // eslint-disable-line comma-dangle
```
Comment (svdgraaf) [inline lib/plugins/aws/deploy/compile/events/apiGateway/tests/methods.js]: @pmuens I couldn't get eslint to parse this correctly, I'm not sure how to fix this correctly without disabling the eslint parser for these lines. Eslint wants me to add comma's here, but when I add them (which don't make sense), it wants me to delete them ;) Perhaps the notation is wrong?

**Pull request closed without merging**